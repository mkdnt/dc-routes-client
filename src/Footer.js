import React, { Component } from 'react'

export class Footer extends Component {
    render() {
        return (
            <div>
                <p>testing footer, what is here?</p>
            </div>
        )
    }
}

export default Footer
