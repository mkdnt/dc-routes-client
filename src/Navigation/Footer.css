.footer {
  background-color: #fff1e6;
  font-size: larger;
  text-align: center;
  padding: 0 0 0 5px;
  border: 2px solid #011328;
  color: #b40421;
  margin: 5px;
}
