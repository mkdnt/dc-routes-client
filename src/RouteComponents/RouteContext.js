import React from 'react';

export default React.createContext({
    routes: [],
    addRoute: () => {},
    editRoute: () => {},
    deleteRoute: () => {},
    editing: null
})