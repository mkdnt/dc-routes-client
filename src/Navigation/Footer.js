import React, { Component } from "react";
import "./Footer.css";

export class Footer extends Component {
  render() {
    return (
      <div className="footer">
        <h1>★★★</h1>
      </div>
    );
  }
}

export default Footer;
