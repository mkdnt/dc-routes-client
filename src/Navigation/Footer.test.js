import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom';
import RouteContext from '../RouteComponents/RouteContext';
import Footer from './Footer';

describe('Footer Component', () => {
  describe('Snapshot', () => {
    it('renders the UI as expected', () => {
      const update = renderer.create(<BrowserRouter><Footer /></BrowserRouter>)
        expect(update).toMatchSnapshot()
    })
  })
  
  describe('Smoke test', () => {
    it('Renders without crashing', () => {
      const div = document.createElement('div');
      
      const value = {
        routes: [],
        addRoute: () => {},
        editRoute: () => {},
        deleteRoute: () => {}
        }

      ReactDOM.render(
        <BrowserRouter>
          <RouteContext.Provider value={value}>
            <Footer />
        </RouteContext.Provider>
        </BrowserRouter>,
      div);
      ReactDOM.unmountComponentAtNode(div);
    })
  })

});