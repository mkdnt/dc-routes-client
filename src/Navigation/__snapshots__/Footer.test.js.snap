// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Snapshot renders the UI as expected 1`] = `
<div
  className="footer"
>
  <h1>
    ★★★
  </h1>
</div>
`;
